### PR TITLE
fix(github-integration): fixed ui and logic of github integration

### PR DESCRIPTION
**Pull request by abhishek-mishra019**
Signed-off-by: Abhishek Mishra <am1426620@gmail.com>

### Functionality:
fixing the UI and logic of Github integration in my-organization and task details page

### Solution:
1. Fixed API trigger for github repo integration.
2. created UI and API  for github PR integration.
3.  created fields in db for projectLink and prLinks.

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
step1. in the my-organisation page, try adding a github repo, two three times and also reload the page to see if it is still integrated.
step 2. in the task details page, try adding a pull request, then see if its status is being shown

### CODE REVIEW

File: functions/model/teams/teams.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ const {getLabelByScope} = require("./tark/getTeamLabels");
 const {deleteLabel} = require("./tark/deleteLabel");
 const {editLabel} = require("./tark/editLabel");
 const {addLabel} = require("./tark/addLabel");
+<<<<<<< HEAD
+const {addProjLink} = require("./tark/addProjLink");
+=======
 const {createDefaultLabels} = require("./tark/createDefaultLabels");
+>>>>>>> 4f45bb16dbee10faf56d3cb7823c87282472ad4e
```
Comment (printf-twinkle) [inline functions/model/teams/teams.js:34]: please check if there is an unresolved merge conflict here

**Review by printf-twinkle — COMMENTED** [review]
please check the comment

File: src/app/body/task-details/git/git.component.ts [hunk 2]
```diff
@@ -18,59 +23,111 @@ export class GitComponent implements OnInit {
...
-        error: (error) => {
-          this.enableLoader = false
-          this.errorHandlerService.showError = true;
-          this.errorHandlerService.getErrorCode(this.componentName, "InternalError", "Api");
-          console.log("Error", error);
-          console.error(error);
-        },
-        complete: () => console.info('Successful')
+    this.teamId = this.taskId.slice(0,3);
+    console.log(this.teamId);
```
Comment (viv9k) [inline src/app/body/task-details/git/git.component.ts:45]: Remove this console log

File: src/app/body/task-details/git/git.component.ts [hunk 3]
```diff
@@ -18,59 +23,111 @@ export class GitComponent implements OnInit {
...
+      this.team=data;
+      console.log(this.team.ProjectLink);
+      this.repoLink=this.team.ProjectLink;
+      if(this.repoLink!=""){
+      this.getPullRequests();
+      }
+      else{
+        this.noRepoLinked=true;
+       }
+    });
```
Comment (viv9k) [inline src/app/body/task-details/git/git.component.ts:67]: This method contains the subscribe to trigger an api, just to get some data, I believe similar data can be sent from parent.

**Review by viv9k — CHANGES_REQUESTED** [review]
The logic for git component can be improved alot

File: functions/model/teams/teams.js (hunk 1 above)
Comment (abhishek-mishra019) [reply functions/model/teams/teams.js:34]: resolved the merge conflict already

File: src/app/body/task-details/git/git.component.ts (hunk 2 above)
Comment (abhishek-mishra019) [reply src/app/body/task-details/git/git.component.ts]: removed

File: src/app/body/task-details/git/git.component.ts (hunk 3 above)
Comment (abhishek-mishra019) [reply src/app/body/task-details/git/git.component.ts]: now discussed, and no need to change anymore

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**